### PR TITLE
Precompute probe row size estimate once

### PR DESCRIPTION
**Pull request by sopel39**
Don't enumerate probe blocks for every appended
join row.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java [hunk 1]
```diff
@@ -179,11 +181,23 @@ private void appendProbeIndex(JoinProbe probe)
         if (previousPosition == position) {
             return;
         }
+        estimatedProbeBlockBytes += getEstimatedProbeRowSize(probe);
+    }
+
+    private int getEstimatedProbeRowSize(JoinProbe probe)
+    {
+        if (estimatedProbeRowSize != -1) {
+            return estimatedProbeRowSize;
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java]: Can you great greatly off estimates when initial data is outlier?

Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java]: the estimate per row is always the same. It was always the same previously too.

`block.getSizeInBytes() / block.getPositionCount();` takes into account entire block size

Comment (findepi) [reply presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java]: i missed the reset call, thanks

File: presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java [hunk 2]
```diff
@@ -179,11 +181,23 @@ private void appendProbeIndex(JoinProbe probe)
...
+        }
+
+        estimatedProbeRowSize = 0;
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java]: i'd prefer to use a local var and populate the class state (field) only when fully calculated.
(one variable more, but makes code easier to reason about)

**Pull request merged**